core/matching/orderbook.go

```diff
@@ -1056,8 +1056,8 @@ func (b *OrderBook) Settled() []*types.Order {
 // GetAllUnParkedPeggedOrderIDs returns the order identifiers of all pegged orders in the order book that are not parked.
 func (b *OrderBook) GetAllUnParkedPeggedOrderIDs() []string {
 	pegged := make([]string, 0, len(b.peggedOrders))
-	for _, o := range b.peggedOrders {
-		if o.Status != vega.Order_STATUS_PARKED {
+	for ID := range b.peggedOrders {
+		if o, ok := b.ordersByID[ID]; ok && o.Status != vega.Order_STATUS_PARKED {
```

Comment:
Well if this is not true, we should probably panic here right? No parked orders should be on the book?